### PR TITLE
CFINSPEC-361: Add `podman_image` resource

### PR DESCRIPTION
**Pull request by ahasunos**
✅ Signed-off-by: Sonu Saha [sonu.saha@progress.com](mailto:sonu.saha@progress.com)

## Related Issue
**CFINSPEC-361: Add `podman_image` resource**

## Description
- This PR adds a new resource: `podman_image`
- Using the`podman_image` resource
Given that the user has called the `podman_image` resource
then the resource allows to test with the container image properties.
- Syntax:
```ruby
  describe podman_image("docker.io/library/busybox") do
      it { should exist }
      its("resource_id") { should eq "sha256:3c19bafed22355e11a608c4b613d87d06b9cdd37d378e6e0176cbc8e7144d5c6" }
  end
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/resources/podman_image.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      o = { image: opts } if opts.is_a?(String)
+      @opts = sanitize_options(o)
+    end
+
+    def image
+      "#{repo}:#{tag}" if object_info.entries.size == 1
+    end
+
+    def repo
+      object_info.repositories[0] if object_info.entries.size == 1
```
Comment (clintoncwolfe) [inline lib/inspec/resources/podman_image.rb]: What happens if object_info.entries.size != 1 ?

File: lib/inspec/resources/podman_image.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      opts
+    end
+
+    def object_info
+      return @info if defined?(@info)
+
+      opts = @opts
+      @info = inspec.podman.images.where do
+        (repository == opts[:repo] && tag == opts[:tag]) || (!id.nil? && !opts[:id].nil? && (id == opts[:id] || id.start_with?(opts[:id])))
+      end
```
Comment (clintoncwolfe) [inline lib/inspec/resources/podman_image.rb]: Check for multiple entries here

File: lib/inspec/resources/podman_image.rb [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    EXAMPLE
+
+    attr_reader :opts, :inspect_info
+
+    def initialize(opts)
+      skip_resource "The `podman_image` resource is not yet available on your OS." unless inspec.os.unix?
+      opts = { image: opts } if opts.is_a?(String)
+      @opts = sanitize_options(opts)
+      @inspect_info = get_inspect_info
```
Comment (Vasu1105) [inline lib/inspec/resources/podman_image.rb]: In `get_inspec_info` method, we actually getting the image data using the `podman image inspect` command, so naming that method to something specific might be helpful. and maybe also the `@inspect_info` to `@image_info`

Comment (ahasunos) [reply lib/inspec/resources/podman_image.rb]: Any suggestion for the method name?

Comment (ahasunos) [reply lib/inspec/resources/podman_image.rb]: get_image_inspect_info?

**Review by Vasu1105 — APPROVED** [review]
Small suggestion else all looks good. Good work!!

File: lib/inspec/resources/podman_image.rb (hunk 3 above)
Comment (ahasunos) [reply lib/inspec/resources/podman_image.rb]: Renamed the attribute and method name to `image_info` and `get_image_info` as discussed.

**Review by Nik08 — APPROVED** [review]
Looks good to me too!

**Comment by Vasu1105** [comment]
@ahasunos Can we have a test that will check if the given image name does not exist?

**Comment by Vasu1105** [comment]
> @ahasunos Can we have a test that will check if the given image name does not exist?

Ignore it seems you already have it.

**Review by dkumaras — COMMENTED** [review]
@ahasunos - Pls chk n cnfrm.

**Comment by Vasu1105** [comment]
@ahasunos I get below result for the non existing image

```
describe podman_image("test") do
  it { should exist }
end

```
```
Profile:   InSpec Profile (podman_test)
Version:   0.1.0
Target:    local://
Target ID: 0744915e-9ac4-5993-a769-a7cf9db10a72

  podman_image test:latest
     ×  Unable to retrieve podman image info for test:latest.
     Error message: Error: inspecting object: unable to inspect "test:latest": failed to find image test:latest: test:latest: image not known
```

I should get false if the image does not exist right? But it's failing with an error.

**Comment by ahasunos** [comment]
@Vasu1105 

Just made the changes. It was failing with an error because I was raising the exception right at the initialisation while finding inspect information for the image.

**Comment by Vasu1105** [comment]
@ahasunos we merged the PR but that fix for the non-existing image will not raise an error in case of Podman connection failures. Please update that in when you are working on the other Podman resources.

**Pull request merged**